src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -99,7 +99,6 @@ private void EnsurePushCapacity()
         public JsonConverter Initialize(Type type, JsonSerializerOptions options, bool supportContinuation)
         {
             JsonTypeInfo jsonTypeInfo = options.GetOrAddJsonTypeInfoForRootType(type);
-            Debug.Assert(options == jsonTypeInfo.Options);
```

Comment:
This assertion has been invalidated by #64646, however it wasn't being caught by our test suite until now.

Reply:
I assume it would still be true if the new comparer introduced in https://github.com/dotnet/runtime/pull/64646 were used.

Reply:
`==` still checks for reference equality, the comparer is internal implementation detail when initializing the caches.